controllers/profiles/common/app_properties.go

```diff
@@ -38,34 +43,313 @@ import (
+	kogitoServiceUrlProtocol                = "http"
+	dataIndexServiceUrlProperty             = "mp.messaging.outgoing.kogito-processinstances-events.url"
+	kafkaSmallRyeHealthProperty             = "quarkus.smallrye-health.check.\"io.quarkus.kafka.client.health.KafkaHealthCheck\".enabled"
+	dataIndexServiceUrlProtocol             = "http"
+	jobServiceURLProperty                   = "mp.messaging.outgoing.kogito-job-service-job-request-events.url"
+	jobServiceKafkaSinkInjectionHealthCheck = `quarkus.smallrye-health.check."org.kie.kogito.jobs.service.messaging.http.health.knative.KSinkInjectionHealthCheck".enabled`
+	jobServiceStatusChangeEventsProperty    = "kogito.jobs-service.http.job-status-change-events"
+	jobServiceStatusChangeEventsURL         = "mp.messaging.outgoing.kogito-job-service-job-status-events-http.url"
+	jobServiceURLProtocol                   = "http"
+	jobServiceDataSourceReactiveURLProperty = "quarkus.datasource.reactive.url"
+
+	imageNamePrefix                          = "quay.io/kiegroup/kogito"
+	DataIndexServiceName                     = "data-index-service"
+	DataIndexName                            = "data-index"
+	JobServiceName                           = "jobs-service"
+	PersistenceTypeEphemeral                 = "ephemeral"
+	PersistenceTypePostgressql               = "postgresql"
 	microprofileServiceCatalogPropertyPrefix = "org.kie.kogito.addons.discovery."
 	discoveryLikePropertyPattern             = "^\\${(kubernetes|knative|openshift):(.*)}$"
+
+	defaultDatabaseName   string = "sonataflow"
+	defaultPostgreSQLPort int    = 5432
+)
+
+var (
+	immutableApplicationProperties = "quarkus.http.port=" + DefaultHTTPWorkflowPortIntStr.String() + "\n" +
+		"quarkus.http.host=0.0.0.0\n" +
+		// We disable the Knative health checks to not block the dev pod to run if Knative objects are not available
+		// See: https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/eventing/consume-produce-events-with-knative-eventing.html#ref-knative-eventing-add-on-source-configuration
+		"org.kie.kogito.addons.knative.eventing.health-enabled=false\n" +
+		"quarkus.devservices.enabled=false\n" +
+		"quarkus.kogito.devservices.enabled=false\n"
+
+	discoveryLikePropertyExpr                    = regexp.MustCompile(discoveryLikePropertyPattern)
+	_                         AppPropertyHandler = &appPropertyHandler{}
 )
 
-var immutableApplicationProperties = "quarkus.http.port=" + DefaultHTTPWorkflowPortIntStr.String() + "\n" +
-	"quarkus.http.host=0.0.0.0\n" +
-	// We disable the Knative health checks to not block the dev pod to run if Knative objects are not available
-	// See: https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/eventing/consume-produce-events-with-knative-eventing.html#ref-knative-eventing-add-on-source-configuration
-	"org.kie.kogito.addons.knative.eventing.health-enabled=false\n" +
-	"quarkus.devservices.enabled=false\n" +
-	"quarkus.kogito.devservices.enabled=false\n"
+type PlatformService interface {
```

Comment:
We are mixing Application properties with object creations in this file. This `PlatformService` must be within the package `platform` and not be exported.

Reply:
Moved inside the`platform` package.